be1-go/hub/standard_hub/message_handling.go

```diff
@@ -193,15 +197,68 @@ func (h *Hub) handleAnswer(senderSocket socket.Socket, byteMessage []byte) error
 		return xerrors.Errorf("query %v already got an answer", answerMsg.ID)
 	}
 
-	channel := h.queries.queries[*answerMsg.ID].Params.Channel
+	catchUpQuery, isCatchupQuery := h.queries.catchupQueries[*answerMsg.ID]
+	_, isGetMessagesByIdQuery := h.queries.getMessagesByIdQueries[*answerMsg.ID]
+
 	*h.queries.state[*answerMsg.ID] = true
 	h.queries.Unlock()
 
+	if isCatchupQuery {
+		h.handleCatchupAnswer(senderSocket, answerMsg, catchUpQuery)
+	} else if isGetMessagesByIdQuery {
+		h.handleGetMessagesByIdAnswer(senderSocket, answerMsg)
+	} else {
+		return xerrors.Errorf("query %v is not catchup or getMessagesById", answerMsg.ID)
+	}
+
+	return nil
+}
+
+func (h *Hub) handleGetMessagesByIdAnswer(senderSocket socket.Socket, answerMsg answer.Answer) {
+	messages := answerMsg.Result.GetMessagesByChannel()
+	for channel, messageArray := range messages {
+		for _, msg := range messageArray {
+
+			var messageData message.Message
+
+			err := json.Unmarshal(msg, &messageData)
+			if err != nil {
+				h.log.Error().Msgf("failed to unmarshal message during getMessagesById answer handling: %v", err)
```

Comment:
To confirm, in this case if a message throws an error when unmarshalled, you still continue and just log the error right?

Reply:
Yes, I handled it almost exactly the same way as the catchup messages but now that you made a comment I'm not sure. @Anghul could you clarify whether there's a reason catchup answers are handled that way ?

Reply:
They were handled that way because when receiving an array of message, maybe one of these messages would fail while unmarshalling, and I wanted the code to be able to still treat the following messages and not just fail in the middle of the array, but then you could want to do it another way dending on your protocol.

Reply:
Thanks, it does make sense to do it that way. I think I'll keep it like this for now @Nazianzenov